app/src/main/java/org/eyeseetea/malariacare/views/question/multiquestion/PhoneMultiQuestionView.java

```diff
@@ -48,9 +56,38 @@ public void setHelpText(String helpText) {
     public void setValue(ValueDB valueDB) {
         if (valueDB != null) {
             mCustomEditText.setText(valueDB.getValue());
+            if (BuildConfig.validationInline) {
+                if (!mCustomEditText.getText().toString().isEmpty()) {
+                        validatePhone(valueDB.getValue());
+                }
+            }
         }
     }
 
+    private void validatePhone(final String value) {
+
+        IPhoneFormatRepository phoneLocalDataSource = new PhoneFormatLocalDataSource();
```

Comment:
The responsibility of a custom view should only be a view but here also search the phone format. I think is more correct configure this view with a format from outside. We would add a Configureformat method and invoke it.